File: 11_simd/problems/00_omp_simd/omp_simd.cpp
/* y[i] = a * x[i] + y[i] (saxpy/axpy) を n 要素について行う */
void saxpy(long n, double a, double * x, double * y) {
  // TODO: 下の for ループの直前に #pragma omp simd を1行追加し, このループをSIMD化せよ.
  for (long i = 0; i < n; i++) {
    y[i] = a * x[i] + y[i];
  }
}

File: 11_simd/problems/00_omp_simd/omp_simd.f90
! y(i) = a * x(i) + y(i) (saxpy/axpy) を n 要素について行う
subroutine saxpy(n, a, x, y)
  implicit none
  integer(8), intent(in) :: n
  real(8), intent(in) :: a
  real(8), intent(in) :: x(n)
  real(8), intent(inout) :: y(n)
  integer(8) :: i
  ! TODO: 下の do ループの直前に !$omp simd を1行追加し, このループをSIMD化せよ.
  do i = 1, n
     y(i) = a * x(i) + y(i)
  end do
end subroutine saxpy
